_posts/2016-04-06-ace-the-technical-interview.markdown

```diff
@@ -0,0 +1,42 @@
+
+Here's a question we commonly ask during technical interviews: 
+> Describe what happens when a person types a URL into their browser and hits 'enter'
+
+This question is great for a number of reasons:
+
+- it's conversational, so the interviewee doesn't have to panic at the dreaded white board marker
+- the answer given can lead to follow-up questions and further discussion
+- the topic is massively practical -- anyone interviewing for an engineering role should understand how the modern internet works
+- it is language/tech stack-agnostic
+
+During my technical interviews for the WeWork engineering team, I was asked this question and here's the answer I gave (kind of):
+
+> "When a user types a URL into the browser and hits 'enter', the browser fairies sweep up all the magical request dust and use their fairy powers to turn it into a beaaaaauuuuutiful gem. Before embarking on their journey, they first look around the fairy village to see if they already have the ingredients they're looking for. If they don't have them, or if the ingredients are stale, the fairies venture out looking for the DNS Fortune Teller, who shows them where to find the Server Wizard. The fairies then bring the request gem to the Server Wizard (along with cookies, because the Server Wizard demands cookies), who sits on a throne of HTML, CSS, and JavaScript. If the Wizard is pleased by the fairies' request gem, he will bellow to the mountains, '200!!!!!!!' (or possibly some variant of '300!!!!!' if he's already seen that particular request gem, but is still in a good mood). If he is displeased by the fairies' cookies or request gem, he will shout angrily '400!!!' (or something in that range), or '500!!! if he's eaten too many cookies and isn't feeling well. 
+
+> If the Wizard accepts the fairy cookies and request gem, they get to return to their fairy village with all sorts of magical wonders to share. Once they arrive home, they put together a fancy party for all their friends, with HTML tables and CSS streamers, and JS magicians to entertain."
+
+So that's basically how the internet works*, and how to pass the technical interview, with some flourish. Stay tuned for a follow up post on how to write any function with Big O(1) while wearing a [silly hat](https://www.youtube.com/watch?v=vEiyBbP28g8)!
+
+[We're hiring!](http://grnh.se/ud2teg) 
+
+*That's not really how the internet works, of course. This is a simplification of what takes place, although the actual processes do involve fairies and a semi-omniscient being sometimes referred to as a Server Wizard. This depiction was written predominantly for the purposes of [comedy](https://youtu.be/iV2ViNJFZC8), but also to foster discussion of how to conduct effective interviews. A "good" answer to this question doesn't necessarily need to touch on every detail, but should be able to reason intelligently and draw conclusions from the discussion. 
```

Comment:
I feel like this takes away from the point of this post, which is...be yourself in an interview.

If you want to fall back on something like this, we could reduce this amount of text significantly 

Thoughts?
```
So that's basically how the internet works and how to pass the technical interview...with some flourish*.

*
server wizard == server
request fairies == client request
etc...

```